vmdb/app/models/orchestration_stack.rb

```diff
@@ -59,4 +59,68 @@ def raw_delete_stack
   def raw_status
     ext_management_system.stack_status(name, ems_ref)
   end
+
+  # Processes tasks received from the UI and queues them
+  def self.process_tasks(options)
+    raise "No ids given to process_tasks" if options[:ids].blank?
+    raise "Unknown task, #{options[:task]}" unless instance_methods.collect(&:to_s).include?(options[:task])
+    options[:userid] ||= "system"
+    invoke_tasks_queue(options)
+  end
+
+  # Performs tasks received from the UI via the queue
+  def self.invoke_tasks(options)
+    local, _remote = partition_ids_by_remote_region(options[:ids])
+    invoke_tasks_local(options.merge(:ids => local)) unless local.empty?
+    # invoke_tasks_remote(options.merge(:ids => remote)) unless remote.empty?
+  end
```

Comment:
@Fryguy can we merge this PR first, so that UI team can start to add the retirement menu for orchestration stacks?

I'll start another PR to refactor the process_tasks as a mixin for `Service`, `OrchestrationStack`, `VmOrTemplate` and `ConfigurationManagerForeman`. The work for `VmOrTemplate` can be a little challenged.